### DISCUSSION TITLE
How to configure macros for Intellisense?

### DISCUSSION THREAD

**Discussion by daldr-ntml** [microsoft/vscode-cpptools#9294] [Q&A] [answered]
Hi, it's a while since I have worked with vscode-cpptools. I want to edit (not build) some C++ source code and I need Intellisense to respond to my #defines to selectively enable code.

What is the recommended way of specifying macros/#defines to vscode-cpptools these days? Should I do so in VS Code Settings (which setting?) or directly in c_cpp_properties.json?

Thanks.

**Comment by sean-mcmanus** [comment, marked as answer]
You can set "defines" in c_cpp_properties.json or use C_Cpp.default.defines ("The value to use in a configuration if defines is not specified, or the values to insert if ${default} is present in defines.").

**Reply by daldr-ntml** [reply]
Thanks
